### ISSUE TITLE
Issue with Failed Payment System Communication

### ISSUE THREAD

**Issue by austinspooner** [SparkDevNetwork/Rock#4185]
### Description

When a scheduled payment fails to process - and it initiates the Failed Payment System Communication in Rock Version: Rock McKinley 10.2 (1.10.2.3) the message does not work and shows a lava error. This bug was introduced with the move to System Communications is 10.3. We are also using the "stock" system communication. 

### Steps to Reproduce

1. Setup a payment that will fail.
2. Wait for the email to get sent.
3. The email will contain a lava merge error.

**Expected behavior:**

The email should contain the correct text.

**Actual behavior:**
This is the message that appears.
![image](https://user-images.githubusercontent.com/657005/79402232-e2f46480-7f50-11ea-86a3-1f4750d3b45e.png)

### Versions

* **Rock Version:** Rock McKinley 10.2 (1.10.2.3)
* **Client Culture Setting:**  en-US

**Comment by MichaelAllen** [comment]
Confirmed on both 10.2 and prealpha (11.0.14).

It appears that the issue is not with the message itself, but with the Rock:CodeEditor control on that page being set to HTML mode instead of Liquid mode. It is seeing the `<` in the lava and trying to convert to a valid HTML tag by adding `=""` to everything after it. 

This: `{% if amount < 0 %}{% assign amount = 0 | Minus:amount %}{% endif %}...` becomes this: `{% if amount < 0="" %}{%="" assign="" amount="0" |="" minus:amount="" %}{%="" endif="" %}=""...`

(This is also a problem in the email template editor, but you are not likely to have a `<` in lava on a template which is why it probably hasn't been noticed before.)

**Comment by austinspooner** [comment]
Is there a quick fix for this or does it need an update?

**Comment by MichaelAllen** [comment]
@austinspooner I'm still digging into it, but unfortunately, it looks like this is probably going to end up being over my level of knowledge to fix.

**Comment by jonedmiston** [comment]
Increased the internal urgency on this.

**Comment by nairdo** [comment]
@austinspooner @MichaelAllen As Michael described, simply opening and saving the SystemCommunication will corrupt the markup.  Mike has fixed that problem (via a34c85e), but restoring your content to the original value is a tiny bit tricky (and we're working out how we might approach this now)...

However, in the mean time, if the corrupt portion of your `Failed Payment` template was stock (original & unaltered), then this is SQL you can manually run that will reset that section back to the correct value (I tested this in my environment several times):

```Sql
IF EXISTS ( SELECT [Id] FROM [SystemCommunication] WHERE [Guid] = '449232B5-9C6B-480E-A881-E317D0BC307E' )
BEGIN
	UPDATE [SystemCommunication] SET
		[Body] = REPLACE( [BODY], 
		-- bad code
		'<a href="" {{="" ''global''="" |="" attribute:''publicapplicationroot''="" }}give""="">{{ ''Global'' | Attribute:''PublicApplicationRoot'' }}Give</a>.
{% else %}
    {% assign amount = Transaction.TotalAmount %}
    {% if amount < 0="" %}{%="" assign="" amount="0" |="" minus:amount="" %}{%="" endif="" %}="" we="" just="" wanted="" to="" make="" you="" aware="" that="" your="" {{="" transaction.transactiontypevalue.value="" |="" downcase="" }}="" payment="" on="" {{="" transaction.transactiondatetime="" |="" date:''m/d/yyyy''="" }}="" in="" the="" amount="" of="" {{="" amount="" |="" formatascurrency="" }}="" did="" not="" process="" successfully.="" {%="" if="" transaction.transactiontypevalue.value="=" ''contribution''="" %}="" if="" you''d="" like,="" you="" can="" re-submit="" your="" contribution="" at=""><a href="" {{="" ''global''="" |="" attribute:''publicapplicationroot''="" }}give""="">{{ ''Global'' | Attribute:''PublicApplicationRoot'' }}Give</a>',
		-- with good code
		'<a href=""{{ ''Global'' | Attribute:''PublicApplicationRoot'' }}Give"">{{ ''Global'' | Attribute:''PublicApplicationRoot'' }}Give</a>.
{% else %}
    {% assign amount = Transaction.TotalAmount %}
    {% if amount < 0 %}{% assign amount = 0 | Minus:amount %}{% endif %}
    We just wanted to make you aware that your {{ Transaction.TransactionTypeValue.Value | Downcase }} payment on {{ Transaction.TransactionDateTime | Date:''M/d/yyyy'' }} in the amount of 
    {{ amount | FormatAsCurrency }} did not process successfully. 
    {% if Transaction.TransactionTypeValue.Value == ''Contribution'' %}
        If you''d like, you can re-submit your contribution at  
        <a href=""{{ ''Global'' | Attribute:''PublicApplicationRoot'' }}Give"">{{ ''Global'' | Attribute:''PublicApplicationRoot'' }}Give</a>.' )
	WHERE [Guid] = '449232B5-9C6B-480E-A881-E317D0BC307E'
END

```

**Comment by nairdo** [comment]
Update: the final data migration to fix any corrupted `Failed Payment` System Communications is fixed via https://github.com/SparkDevNetwork/Rock/commit/6a70772d0986abac4c81194803039fa27f41cd40 and 609942e9.
